src/main/java/com/mongodb/jdbc/MongoConnection.java

```diff
@@ -66,7 +66,12 @@ public abstract class MongoConnection implements Connection {
     private static Map<String, FileHandler> fileHandlers = new HashMap<String, FileHandler>();
     private String logDirPath;
 
-    public MongoConnection(ConnectionString cs, String database, Level logLevel, File logDir) {
+    public MongoConnection(
+            ConnectionString cs,
+            String database,
+            Level logLevel,
+            File logDir,
+            String[] clientInfo) {
```

Comment:
If we keep adding more and more connection properties to MongoConnection maybe we'll want to have a dedicated data structure for it. It seems that this is still Ok right now, but if you think it makes sense to switch to a data structure (a simple POJO will do), feel free to do it.

Reply:
I added a POJO MongoConnectionProperties to address this.